river/cluster/textclust.py

```diff
@@ -0,0 +1,595 @@
+import math
+import typing
+
+import numpy as np
+import pandas as pd
+from sklearn.cluster import AgglomerativeClustering
```

Comment:
An issue here is that scikit-learn is not one of River's dependencies, and we would like to keep it like that. Agglomerative clustering is not a difficult algorithm to implement, so maybe we could implement a version ourselves. Also that might work better with the fact that we're using dictionaries.

Reply:
The question here is, where do we place this implementation? Utils? Also, in what detail should this be implemented: 

1) Textclust provides a pre-computed distance matrix. It is possible to implement a version that only uses this mode. Otherwise, it would be possible to additionally allow different distance metrics (e.g., euclidean, cosine, etc.). 

2) Which linkage modes should we offer (single, complete average?)

Overall we should decide if we only treat agglomerative clustering as util for textclust or if we should implement it with the intention to be used by other algorithms as well.